src/lib/markbind/src/preprocessors/variablePreprocessor.js

```diff
@@ -0,0 +1,421 @@
+      }
+    });
+
+    return { pageImportedVariables, pageVariables };
+  }
+
+  /**
+   * Extracts variables specified as <include var-xx="..."> in include elements.
+   * @param includeElement to extract inline variables from
+   */
+  static extractIncludeInlineVariables(includeElement) {
+    const includeInlineVariables = {};
+
+    Object.keys(includeElement.attribs).forEach((attribute) => {
+      if (!attribute.startsWith('var-')) {
+        return;
+      }
+      const variableName = attribute.slice(4);
+      includeInlineVariables[variableName] = includeElement.attribs[attribute];
+    });
+
+    return includeInlineVariables;
+  }
+
+  /**
+   * Extracts variables specified as <variable> in include elements.
+   * @param includeElement to search child nodes for
+   */
+  static extractIncludeChildElementVariables(includeElement) {
+    if (!(includeElement.children && includeElement.children.length)) {
+      return {};
+    }
+    const includeChildVariables = {};
+
+    includeElement.children.forEach((child) => {
+      if (child.name !== 'variable' && child.name !== 'span') {
+        return;
+      }
+      const variableName = child.attribs.name || child.attribs.id;
+      if (!variableName) {
+        logger.warn(`Missing 'name' or 'id' in variable for ${includeElement.attribs.src}'s include in ${
+          includeElement.attribs[ATTRIB_CWF]}.\n`);
+        return;
+      }
+      if (!includeChildVariables[variableName]) {
+        includeChildVariables[variableName] = cheerio.html(child.children);
+      }
+    });
+
+    return includeChildVariables;
+  }
+
+
+  /**
+   * Extract variables from an <include> element.
+   * See https://markbind.org/userGuide/reusingContents.html#specifying-variables-in-an-include for usage.
+   * It is a subroutine for {@link renderIncludeFile}
+   * @param includeElement include element to extract variables from
+   */
+  static extractIncludeVariables(includeElement) {
+    const includeInlineVariables = VariablePreprocessor.extractIncludeInlineVariables(includeElement);
+    const includeChildVariables = VariablePreprocessor.extractIncludeChildElementVariables(includeElement);
+
+    return {
+      ...includeChildVariables,
+      ...includeInlineVariables,
+    };
+  }
+
+
+  /*
+   * --------------------------------------------------
+   * Combined (both site and page variables) rendering methods
+   */
+
+
+  /**
+   * Renders an <include> file with the supplied context, returning the rendered
+   * content and new context with respect to the child content.
+   * @param filePath of the included file source
+   * @param node of the include element
+   * @param context object containing the parent <include> (if any) variables for the current context,
+   *        which have greater priority than the extracted include variables of the current context.
+   * @param asIfAt where the included file should be rendered from
+   */
+  renderIncludeFile(filePath, node, context, asIfAt) {
```

Comment:
The gist of the revamp is summarised in this method and the one below as mentioned.
The main highlights are:
- the process is made pure - no static `Parser` variables are needed any longer
- no hacky workarounds like `Proxy` due to removal of the 2-step extraction/rendering into one
- better performance 😁 

The differences in this and render page are:
- we need to render `<include>` files with the `<include **var-xx="..."**>**<variable>...</variable>**</include>`s
- the child context is returned for processing of nested `<include>`s